project/JavalibIRCleaner.scala

```diff
@@ -218,6 +235,15 @@ final class JavalibIRCleaner(baseDirectoryURI: URI) {
       implicit val pos = tree.pos
 
       tree match {
+        // new AnonFunctionN(closure)  -->  closure
+        case New(AnonFunctionNClass(n), _, List(closure)) =>
+          closure
+
+        // someFunctionN.apply(args)  -->  someFunctionN(args)
+        case Apply(ApplyFlags.empty, fun, MethodIdent(FunctionApplyMethodName(n)), args)
+            if isFunctionNType(n, fun.tpe) =>
+          JSFunctionApply(fun, args)
```

Comment:
Perhaps something that's also worth pointing out is that these two cases are the only transformation that is truly specific to the function rewriting. In the rest of #4677, there will also be rewrites for Tuples and for `scala.runtime.*Ref` types. These will reuse all of the infrastructure set up in this commit, so that they will do nothing but add new pairs in `ClassNameSubstitutions`.

Reply:
Ah, yes, that makes a difference. I'll have a closer look at #4677 then.